atomic_reactor/auth.py

```diff
@@ -146,12 +173,15 @@ def __call__(self, request):
         url_parts = urlparse(request.url)
 
         if self.V1_URL.search(url_parts.path):
-            if not self.username or not self.password:
+            if (not self.username or not self.password) and not self.auth_b64:
                 # V1 API only supports basic auth which requires user/pass
                 return request
 
             if not self.v1_auth:
-                self.v1_auth = HTTPBasicAuth(self.username, self.password)
+                if self.auth_b64:
```

Comment:
Could we use this logic just one?

```python
if self.auth_b64:
    basic_auth = HTTPBasicAuthWithB64(self.auth_b64)
elif self.username and self.password:
    basic_auth =  HTTPBasicAuth(self.username, self.password)
else:
    basic_auth = None

...

### V1
if not self.v1_auth and basic_auth:
    self.v1_auth = basic_auth

### V2
....
if basic_auth:
    self.v2_auths.append(basic_auth)

```

Reply:
Nice catch. Thanks!
Done. Also rebased on master.